app/src/parser.c

```diff
@@ -193,6 +193,37 @@ parser_error_t parser_printArgument(const flow_argument_list_t *v,
     return parser_ok;
 }
 
+
+parser_error_t parser_printArgumentOptionalDelegatorID(const flow_argument_list_t *v,
+                                               uint8_t argIndex, char *expectedType, jsmntype_t jsonType,
+                                               char *outVal, uint16_t outValLen,
+                                               uint8_t pageIdx, uint8_t *pageCount) {
+    MEMZERO(outVal, outValLen);
+
+    if (argIndex >= v->argCount) {
```

Comment:
why can't we pass `parser_context_t* ctx` instead of `v` and `argIndex` ?

Reply:
This is for consistency with surrounding code. There are several functions of this kind. They should be consistent. If this issue is significant, I can look at it and modify it everywhere (if possible), but then I would put this refactor into a separate request.

Reply:
Oh, now I see. Function above gets `v` and `argIndex`, function below gets `argumentCtx`. So much for consistency :shrug: Anyway, we should definitely consider some subsequent refactoring whether it wouldn't make sense to move the responsibility